AudioPlayer/src/main/kotlin/com.github.diamondminer88.plugins/OggMetadataFetcher.kt

```diff
@@ -0,0 +1,110 @@
+package com.github.diamondminer88.plugins
+
+import java.io.ByteArrayOutputStream
+import java.net.HttpURLConnection
+import java.net.URL
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+object OggMetadataFetcher {
+    data class OggMetadata(
+        val codec: String,
+        val sampleRate: Int,
+        val granulePos: Long,
+        val duration: Double
+    )
+
+    @JvmStatic
+    fun fetch(url: String): OggMetadata? {
+        var sampleRate = -1
+        var codec = ""
+        val head = fetchBytes(url, 0, 32767) ?: return null
```

Comment:
Is 32KiB really necessary? I'm not familiar with these codecs but make this as little as necessary.